website/docs/apis/java-client.md

```diff
@@ -0,0 +1,292 @@
+```java
+admin.dropTable(TablePath.of("test_db", "test_table"), true).get();
+```
+
+### Table Exists
+```java
+admin.tableExists(tablePath).get();
+```
+
+## Managing Partitions
+### Create Partition
+```java
+admin.createPartition(tablePath, new PartitionSpec(Collections.singletonMap("age", "10")).get();
+```
+
+### List Partitions
+```java
+List<PartitionInfo> partitions = admin.listPartitionInfos(TablePath.of("fluss", "test_table")).get();
+partitions.forEach(System.out::println);
+```
+
+### Drop Partition
+```java
+admin.dropPartition(tablePath, new PartitionSpec(Collections.singletonMap("age", "10")).get();
+```
+
+## Retrieving Metadata
+### Get Latest KV Snapshots
+```java
+KvSnapshots snapshots = admin.getLatestKvSnapshots(tablePath).get();
+System.out.println(snapshots.getSnapshotId(1));
+System.out.println(snapshots.getBucketIds());
+System.out.println(snapshots.getLogOffset(1));
+System.out.println(snapshots.getTableId());
+System.out.println(snapshots.getPartitionId());
+```
+
+### Get KV Snapshot Metadata
+```java
+KvSnapshotMetadata snapshotMetadata = admin.getKvSnapshotMetadata(tableBucket, snapshotId.getAsLong()).get();
+```
+
+### Get Server Nodes
+```java
+List<ServerNode> serverNodes = admin.getServerNodes().get();
+```
+
+
+## Writers
+In order to write data to Fluss tables, first you need to create a Table instance.
+```java
+TablePath tablePath = TablePath.of("fluss", "users");
+Table table = connection.getTable(tablePath);
+```
+
+In Fluss we have both Primary Key tables and Log Tables, so the client provides different functionality depending on the table type.
+You can use an `UpsertWriter` to write data to a Primary Key table, and an `Append` to write data to a Log Table.
+````java
+table.newUpsert().createWriter();
+table.newAppend().createWriter();
+````
+
+Let's take a look at how to write data to a Primary Key table.
+```java
+List<User> users = List.of(
+        new User("1", 20, LocalDateTime.now() , true),
+        new User("2", 22, LocalDateTime.now() , true),
+        new User("3", 23, LocalDateTime.now() , true),
+        new User("4", 24, LocalDateTime.now() , true),
+        new User("5", 25, LocalDateTime.now() , true)
+);
+```
+
+**Note:** Currently data in Fluss is written in the form of `rows`, so we need to convert our POJO to `GenericRow`, while the Fluss community is working to provide
+a more user-friendly API for writing data.
+```java
+Table table = connection.getTable(tablePath);
+
+List<GenericRow> rows = users.stream().map(user -> {
+    GenericRow row = new GenericRow(4);
+    row.setField(0, BinaryString.fromString(user.getId()));
+    row.setField(1, user.getAge());
+    row.setField(2, TimestampNtz.fromLocalDateTime(user.getCreatedAt()));
+    row.setField(3, user.isActive());
+    return row;
+}).collect(Collectors.toList());
+        
+System.out.println("Upserting rows to the table");
+rows.forEach(row -> {     
+    try {
+        table.newUpsert().createWriter().upsert(row).get();
```

Comment:
Usually, we should reuse the writer, and `flush()` after writing a batch of records instead of blocking `get()` for every `upsert()`.